src/factories/calorimetry/CalorimeterClusterRecoCoG_factoryT.h

```diff
@@ -48,15 +53,16 @@ class CalorimeterClusterRecoCoG_factoryT :
         // SpdlogMixin logger initialization, sets m_log
         InitLogger(app, GetPrefix(), "info");
 
-        // Algorithm configuration
-        auto cfg = GetDefaultConfig();
+        app->RegisterParameter(param_prefix + ":samplingFraction", m_algo.getProperty<double>("samplingFraction"));
+        app->RegisterParameter(param_prefix + ":logWeightBase", m_algo.getProperty<double>("logWeightBase"));
+        app->RegisterParameter(param_prefix + ":energyWeight", m_algo.getProperty<std::string>("energyWeight"));
+        app->RegisterParameter(param_prefix + ":enableEtaBounds", m_algo.getProperty<bool>("enableEtaBounds"));
 
-        app->SetDefaultParameter(param_prefix + ":samplingFraction", cfg.sampFrac);
-        app->SetDefaultParameter(param_prefix + ":logWeightBase", cfg.logWeightBase);
-        app->SetDefaultParameter(param_prefix + ":energyWeight", cfg.energyWeight);
-        app->SetDefaultParameter(param_prefix + ":enableEtaBounds", cfg.enableEtaBounds);
+        m_algo.setProperty("samplingFraction", app->GetParameterValue<double>(param_prefix + ":samplingFraction"));
+        m_algo.setProperty("logWeightBase", app->GetParameterValue<double>(param_prefix + ":logWeightBase"));
+        m_algo.setProperty("energyWeight", app->GetParameterValue<std::string>(param_prefix + ":energyWeight"));
+        m_algo.setProperty("enableEtaBounds", app->GetParameterValue<bool>(param_prefix + ":enableEtaBounds"));
```

Comment:
This is obviously ugly. It should also be a loop over defined properties. What prevents it from working is that SetDefaultParameter takes a `T& val` that `m_algo.getProperty<T>` can't satisfy. Or at least I didn't figure out yet how to get it to satisfy.

Reply:
https://github.com/eic/EICrecon/pull/1154/commits/b14d3dcfa9c24754922b5223e168501a1af057c7

Reply:
Problem here was that JANA2 doesn't support std::vector<bool> (but instead std::array<T>) and the visit with a variant that includes std::vector<bool> trips up. We if constexpr our way around this now, but warn if we hit this situation.